pennylane_lightning/src/util/LinearAlgebra.hpp

```diff
@@ -745,4 +748,69 @@ inline auto matrixMatProd(const std::vector<std::complex<T>> m_left,
 
     return m_out;
 }
+
+/**
+ * @brief Calculate the squared norm of a vector
+ */
```

Comment:
This function could probably be templated for pointers and vectors.
I think it may be useful to add docstrings args too.

Reply:
I agree, I will add them!